src/CSFML/Graphics/Shader.cpp

```diff
@@ -419,22 +418,22 @@ void sfShader_setTextureParameter(sfShader* shader, const char* name, const sfTe
     assert(shader);
     assert(name);
     assert(texture);
-    shader->This.setUniform(name, *texture->This);
+    shader->setUniform(name, *texture->This);
 }
 
 
 ////////////////////////////////////////////////////////////
 unsigned int sfShader_getNativeHandle(const sfShader* shader)
 {
     assert(shader);
-    return shader->This.getNativeHandle();
+    return shader->getNativeHandle();
 }
 
 
 ////////////////////////////////////////////////////////////
 void sfShader_bind(const sfShader* shader)
 {
-    sf::Shader::bind(shader ? &shader->This : nullptr);
+    sf::Shader::bind(shader ? static_cast<const sf::Shader*>(shader) : nullptr);
```

Comment:
the static cast is unnecessary and the ternary is also unnecessary

the implicit Derived to Base conversion is enough.